actix-files/src/lib.rs

```diff
@@ -901,4 +904,40 @@ mod tests {
         let bytes = test::read_body(resp).await;
         assert!(format!("{:?}", bytes).contains("/tests/test.png"));
     }
+
+    #[actix_rt::test]
+    async fn test_path_filter() {
+        // prevent searching subdirectories
+        let st = Files::new("/", ".")
+            .path_filter(|path, _| path.components().count() == 1)
+            .new_service(())
+            .await
+            .unwrap();
+
+        let req = TestRequest::with_uri("/Cargo.toml").to_srv_request();
+        let resp = test::call_service(&st, req).await;
+        assert_eq!(resp.status(), StatusCode::OK);
+
+        let req = TestRequest::with_uri("/tests/test.png").to_srv_request();
+        let resp = test::call_service(&st, req).await;
+        assert_eq!(resp.status(), StatusCode::NOT_FOUND);
```

Comment:
This test doesn't really prove that the filter works, it would be the same outcome if the file really wasnt there. If we set a `.default_handler()` that returns a different status code then the test would be valid.

Reply:
The default handler is calso called when the file doesn't exist. maybe use 'src/lib.rs' as a test path?

Reply:
true; that would be a better test case at lease